examples/Handler Methods.ipynb

```diff
@@ -0,0 +1,717 @@
+    "4. _did_fit(data_container, context): Apply side effects after fit.\n",
+    "\n",
+    "### handle_transform\n",
+    "\n",
+    "1. _will_process(data_container, context): Apply side effects before any step method.\n",
+    "2. _will_transform(data_container, context): Apply side effects before transform.\n",
+    "3. _transform_data_container(data_container, context): Fit transform data container.\n",
+    "4. _did_transform(data_container, context): Apply side effects after transform. \n",
+    "5. _did_process(data_container, context): Apply side effects after any step method.\n",
+    "\n",
+    "### When to use handler methods ? \n",
+    "\n",
+    "When you need to apply side effects, or change the execution flow:\n",
+    "\n",
+    "- Edit the DataContainer\n",
+    "- Call a method on a step\n",
+    "- Mini-Batching\n",
+    "- Caching\n",
+    "- etc.\n",
+    "\n",
+    "### HandleOnlyMixin\n",
+    "\n",
+    "Inherit from HandleOnlyMixin, to only implement the handler methods, and forbid implementing fit or transform or fit_transform without the handles.\n",
+    "\n",
+    "### ForceHandleMixin\n",
+    "\n",
+    "Inherit from ForceHandleMixin, to automatically calls handle methods in the transform, fit, and fit_transform methods. A step might have to be forced to pass through the lifecycle methods. This is true for all of the steps that can be called from the outside world. For example, you might want to transform all of the data inside the data container. \n",
+    "\n",
+    "### ForceHandleOnlyMixin\n",
+    "\n",
+    "Inherit from ForceHandleOnlyMixin to require the implementation of handler methods, AND automatically call handle methods in the transform, fit, and fit_transform methods.\n",
+    "\n",
+    "\n",
+    "## Examples\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### ToNumpy"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from neuraxle.base import BaseStep, DataContainer, ExecutionContext, ForceHandleMixin, HandleOnlyMixin, TruncableSteps\n",
+    "import numpy as np\n",
+    "\n",
+    "class ToNumpy(ForceHandleMixin, BaseStep):\n",
+    "    \"\"\"\n",
+    "    Convert data inputs, and expected outputs to a numpy array.\n",
+    "    \"\"\"\n",
+    "\n",
+    "    def _will_process(self, data_container: DataContainer, context: ExecutionContext) -> (DataContainer, ExecutionContext):\n",
+    "        return data_container.to_numpy(), context"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Transform Expected Outputs\n",
+    "Consider a wrapper step that would transform the expected outputs instead of the data inputs.\n",
+    "\n",
+    "Create such a step in 4 easy steps (toudoum tish) : \n",
+    "\n",
+    "1. Create a step that inherits from **ForceHandleOnlyMixin**, and **MetaStepMixin**."
```

Comment:
The numbering doesn't work and restarts for the points "2", "3" and so forth below, because those markdown lists are separated. I suggest replacing with a `- `. 
```suggestion
    "- Create a step that inherits from **ForceHandleOnlyMixin**, and **MetaStepMixin**."
```

Note: you mention above `Create a step that inherits from [...]`. The reader might be tempted to code it by himself again. We should make it more clear that this steps, and the other ones discussed below, already exists and that the present page is really just an example of how someone could redo it. 

A solution could be to use the links to the other parts in the documentation as I've suggested above, to link to the existing documentation classes.